src/views/MarketFilter.tsx

```diff
@@ -46,36 +46,42 @@ export const MarketFilter = ({
   hideNewMarketButton,
   compactLayout = false,
   searchPlaceholderKey = STRING_KEYS.MARKET_SEARCH_PLACEHOLDER,
-  shouldHideUnlaunchedMarkets,
-  onShouldHideUnlaunchedMarketsChange,
 }: MarketFilterProps) => {
   const stringGetter = useStringGetter();
   const navigate = useNavigate();
+  const dispatch = useAppDispatch();
   const { hasPotentialMarketsData } = usePotentialMarkets();
   const { uiRefresh, pml: showLaunchMarkets } = testFlags;
   const showProposeButton = hasPotentialMarketsData && !hideNewMarketButton;
+  const shouldHideLaunchableMarkets = useAppSelector(getShouldHideLaunchableMarkets);
+
+  const onShouldHideLaunchableMarkets = useCallback(
+    (shouldHide: boolean) => {
+      dispatch(setShouldHideLaunchableMarkets(!shouldHide));
```

Comment:
does onCheckedChange really provide the old value?

Reply:
it does. its an `onChange` event so I have to invert in order for it to update

Reply:
Kind of wonky behavior honestly

Reply:
surely this should just be `shouldHide` instead of `!shouldHide`

Reply:
oh I see

Reply:
If you use `shouldHide` then the redux dispatch will send in `false` since the state doesn't change